pkg/templatelookup/mandatory.go

```diff
@@ -25,11 +26,72 @@ func GetMandatory(ctx context.Context, kymaClient client.Reader) (ModuleTemplate
 	mandatoryModules := make(map[string]*ModuleTemplateInfo)
 	for _, moduleTemplate := range mandatoryModuleTemplateList.Items {
 		if moduleTemplate.DeletionTimestamp.IsZero() {
-			mandatoryModules[moduleTemplate.Name] = &ModuleTemplateInfo{
-				ModuleTemplate: &moduleTemplate,
+			currentModuleTemplate := &moduleTemplate
+			moduleName := GetModuleName(currentModuleTemplate)
+			if mandatoryModules[moduleName] != nil {
+				var err error
+				currentModuleTemplate, err = GetDesiredModuleTemplateForMultipleVersions(currentModuleTemplate,
+					mandatoryModules[moduleName].ModuleTemplate)
+				if err != nil {
+					mandatoryModules[moduleName] = &ModuleTemplateInfo{
+						ModuleTemplate: nil,
+						Err:            err,
+					}
+					continue
+				}
+			}
+			mandatoryModules[moduleName] = &ModuleTemplateInfo{
+				ModuleTemplate: currentModuleTemplate,
 				Err:            nil,
 			}
 		}
 	}
 	return mandatoryModules, nil
 }
+
+// TODO: Create an issue to remove this function and only use the spec.ModuleName when mandatory modules use modulectl
+func GetModuleName(moduleTemplate *v1beta2.ModuleTemplate) string {
```

Comment:
will create issues to remove the fallback after warden ModuleTemplate is created using modulectl